sshuttle/server.py

```diff
@@ -162,14 +162,44 @@ def _log(self):
         udp_channels = self.get_and_reset_max_udp_channels()
         handlers = self.get_and_reset_max_handlers() - (dns_channels + udp_channels)
 
-        log('handlers %d, dns_channels %d, upd_channels %d, mux_size %d, mux_outbufs %d, mux_flush_count %d, '
+        outbufs = self.get_and_reset_max_outbufs()
+        fullness = self.get_and_reset_max_fullness()
+        too_full_flush_count = self.get_and_reset_too_full_flush_count()
+        mux_wrapper_buffer_size = self.get_and_reset_size_to_edge()
+        paused_to_the_edge_count = self.get_and_reset_paused_to_edge_count()
+        executed_read_count = self.get_and_reset_executed_read_count()
+        executed_write_count = self.get_and_reset_executed_write_count()
+        select_count = self.get_and_reset_select_count()
+        mux_not_ready_for_read_count = self.get_and_reset_mux_not_ready_for_read_count()
+        mux_not_ready_for_write_count = self.get_and_reset_mux_not_ready_for_write_count()
+
+        # This is needed because the dns and upd channels are on a timer. Timing might make the max dns and udp to
+        # be larger than handlers.
+
+        if handlers < 0:
+            handlers = 0
+
+        # We manipulate these values.  Just guarding against mistakes.
```

Comment:
maybe worth a log of some sort in these cases since a mistake likely means some of our data is off and therefore less reliable